src/ui/OwlUI.ts

```diff
@@ -47,28 +53,56 @@ export class OwlUI {
         };
     }
     /**
-     * Mount all of a plugin's Components
+     * Mount all of a plugin's Components to Owl and register its components'
+     * intents, actions and commands (TODO: this does not work for
+     * sub-components since those are mounted by their parent and we do not have
+     * access to their instantiation).
      *
      * @param {JWOwlUIPlugin} pluginInstance
      * @returns {Promise<void>}
      */
-    async _mountPluginComponents(pluginInstance: JWOwlUIPlugin): Promise<void> {
+    async _handleComponents(pluginInstance: JWOwlUIPlugin): Promise<void> {
         const env: PluginEnv = await this._createPluginEnv(pluginInstance);
-        const components = pluginInstance.componentsRegistry.slice();
-        components.forEach((_Component: typeof Component) => {
-            this._mount(_Component, env);
-        });
+        const Components = pluginInstance.componentsRegistry.slice();
+        for (let i = 0; i < Components.length; i++) {
+            const component = new Components[i](env);
+            await this._mount(component);
+            this._register(component, pluginInstance);
+        }
```

Comment:
Why do you need the slice ? Why not the following:
```
plugin.componentsRegistry.forEach(Component => {
    const component = new Component(env);
    await this._mount(component);
    this._register(component, plugin);
});
```

Reply:
I don't anymore. Also there is no `await` in the loop so I can put a `forEach` again.